lib/arborist/reify.js

```diff
@@ -153,16 +156,62 @@ module.exports = cls => class Reifier extends cls {
       return this[_submitQuickAudit]()
     }
 
+    // ok, we're about to start touching the fs.  need to roll back
+    // if we get an early termination.
+    let reifyTerminated = null
+    const removeHandler = onExit(({signal}) => {
+      // only call once.  if signal hits twice, we just terminate
+      removeHandler()
+      reifyTerminated = Object.assign(new Error('process terminated'), {
+        signal,
+      })
+      return false
+    })
+
+    let rollback = _rollbackRetireShallowNodes
+
```

Comment:
Well, it would be much _more_ repetitive to check along the way through each of these steps.  This function could be DRY'ed up a bit, but only by making it a bit more convoluted to read.

Yes, I'll add a comment, though 😅